ariadne/cardano/src/Ariadne/Wallet/Backend/KeyStorage.hs

```diff
@@ -455,6 +456,26 @@ checkWalletsWithoutSecretKey pwl confirmation = do
     whenM (confirmation $ ConfirmDelBrokenWallets $ T.dropEnd 1 $ walletsWithoutKeysText) $
       mapM_ (pwlDeleteWallet pwl) rootIDsWithMissedSecretKeys
 
+changePassword
+  :: PassiveWalletLayer IO
+  -> WalletFace
+  -> IORef (Maybe WalletSelection)
+  -> IO PassPhrase
+  -> (WalletReference -> IO PassPhrase)
+  -> WalletReference
+  -> IO ()
+changePassword pwl WalletFace{..} walletSelRef getPassTemp getPassPhrase walletRef = do
+  mWalletSel <- readIORef walletSelRef
+  case mWalletSel of
+    Nothing -> pass
+    Just selection -> case selection of
+        WSRoot hdrId -> do
+          newPassword <- getPassTemp
+          oldPassword <- getPassPhrase walletRef
+          throwLeftIO $ pwlUpdateWalletPassword pwl hdrId newPassword oldPassword
```

Comment:
1) Makes sense
2) Am I getting right, that your idea is to start `changePassword` dialog similar to `insertPassword` (not by button click, but by backend event)?

Reply:
Well, it can be one dialog with some parameters. Just in case, there should be two modes: 
1. When we change password or create a new wallet.
2. When we enter password of existing wallet.

Differences are:
1. There should be some text which makes it clear which password is expected (new password or old password).
2. In the first mode password should be requested twice (confirmation).

> not by button click, but by backend event

Yes. Moreover, I would consider having only one way to request password, but that's a separate issue: https://serokell.slack.com/archives/C9B1DAMDJ/p1545428962005100

Reply:
I still don't understand the first part (about `walletRef`). It's not longer passed here, but the logic still doesn't look correct.
This code unconditionally changes the password of the selected wallet. But `getPassPhrase` from this file unconditionally passes that `walletRef` (which is the command's argument) to `getPassPhrase` from another file. What if `walletRef` is not the same as the selected wallet? Why do we even need this `walletRef`? Why not `WalletRefSelection`?

Reply:
Probably, misunderstood your comment about `walletRef`:)

Reply:
Looks reasonable now.